students/TinaB/lesson03/strformat_lab.py

```diff
@@ -0,0 +1,99 @@
+
+# Task Three ----------------------
+"""
+Dynamically Building up Format Strings
+Rewrite: "the 3 numbers are: {:d}, {:d}, {:d}".format(1,2,3)
+to take an arbitrary number of values.
+ """
+print("Task Three:")
+
+def formatter(in_tuple):
+    """ function to print input tuple """
+    formatted_string = "the numbers are: "
+    form_string = "{:d}, " * (len(in_tuple)-1) + "{:d}"
+    formatted_string = "the " + \
+        str(len(in_tuple)) + " numbers are: " + form_string
+    return formatted_string.format(*in_tuple)
+
+
+print(formatter((2, 3, 5, 7, 9)))
+
+# Task Four ----------------------
+"""
+Given a 5 element tuple: ( 4, 30, 2017, 2, 27)
+use string formating to print: '02 27 2017 04 30'
+Hint: use index numbers to specify positions."""
+
+print("Task Four:")
+input_tuple = (4, 30, 2017, 2, 27)
+# test print to match:
+print("02 27 2017 04 30")
+print("{:02} {:02} {} {:02} {:02}".format(
+    input_tuple[3], input_tuple[4], input_tuple[2], input_tuple[0], input_tuple[1]))
+print(f"{input_tuple[3]:02} {input_tuple[4]:02} {input_tuple[2]} {input_tuple[0]:02} {input_tuple[1]:02}")
+
+# Task Five -----------------------
+"""
+Given the following four element list:['oranges', 1.3, 'lemons', 1.1] 
+Write an f-string that will display: The weight of an orange is 1.3 and the weight of a lemon is 1.1
+Now see if you can change the f-string so that it displays the names of the fruit in upper case, 
+and the weight 20% higher.
+"""
+print("Task Five:")
+element_list = ['oranges', 1.3, 'lemons', 1.1]
+# String to match:
+print("The weight of an orange is 1.3 and the weight of a lemon is 1.1")
+# part one match test string
+print(f"The weight of an {element_list[0][:-1]} is {element_list[1]} and the weight of a {element_list[2][:-1]} is {element_list[3]}")
+# part two - print fruit upper case
+print(f"The weight of an {element_list[0][:-1].upper()} is {element_list[1]} and the weight of a {element_list[2][:-1].upper()} is {element_list[3]}")
+# part three - print weight 20% higher.
+print(f"The weight of an {element_list[0][:-1].upper()} is {element_list[1]*1.20} and the weight of a {element_list[2][:-1].upper()} is {element_list[3]*1.20}")
+
+# Task Six -----------------------
+"""
+Write some Python code to print a table of several rows, each with a name, an age and a cost. 
+Make sure some of the costs are in the hundreds and thousands to test your alignment specifiers.
+"""
+print("Task Six:")
+
+table_list = [('Carina', 27, 15, 156.25), ('Timothy', 2, 127.63),
+              ('Paul', 12, 1758.17), ('Jeffery', 65, 199), ('Ruben', 107, 165, 248.25),
+              ('Tristan', 85, 8, 156.92), ('Shannon', 44, 556.2), ('Elizabeth', 15, 166, 156.25),
+              ('Tom', 39, 825), ('Tricia', 93, 156), ('Allegra', 68, 962.30)]
+
+# print table header:
+print('{:<10}{:<6}{:>12}'.format('Name', 'Age', 'Cost'))
+# table
+for item in table_list:
+    print('{:<12s}{:<6}{:>10,.2f}'.format(*item))
```

Comment:
The alignment on the second column looks a bit off here. Numbers are generally right-aligned in spreadsheets etc.